extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildBufferAggregator.java

```diff
@@ -53,6 +55,14 @@
   private final IdentityHashMap<ByteBuffer, Int2ObjectMap<HllSketch>> sketchCache = new IdentityHashMap<>();
   private final Striped<ReadWriteLock> stripedLock = Striped.readWriteLock(NUM_STRIPES);
 
+  /**
+   * Used by {@link #init(ByteBuffer, int)}. We initialize by copying a prebuilt empty sketch object.
+   * {@link HllSketchMergeBufferAggregator} does something similar, but we don't share code to create emptySketch
+   * because it is not exactly the same (the "build" flavor initializes based on tgtHllType, but the "merge" flavor
+   * always initializes using HLL_8).
+   */
```

Comment:
This is not so. HllSketchMergeBufferAggregator initializes HllUnion objects as opposed to HllSketch objects here. And HllUnion constructor does not even have the target type parameter (it is always in HLL_8 mode, but it is an implementation detail). The target type is used when the final result is obtained from the union in get().

Reply:
Thanks for clarifying. I just read the javadocs for Union and HllSketch and the situation is more clear. I adjusted the comment to the following,

```java
  /**
   * Used by {@link #init(ByteBuffer, int)}. We initialize by copying a prebuilt empty Union image.
   * {@link HllSketchBuildBufferAggregator} does something similar, but different enough that we don't share code. The
   * "build" flavor uses {@link HllSketch} objects and the "merge" flavor uses {@link Union} objects.
   */
```